content/posts/2023-11-20-tools-for-learning-llvm-tablegen.md

```diff
@@ -0,0 +1,433 @@
+---
+author: "David Spickett"
+date: "2023-11-20"
+tags: ["TableGen"]
+title: "Tools for Learning LLVM TableGen"
+---
+
+[TableGen](https://github.com/llvm/llvm-project/tree/main/llvm/utils/TableGen)
+is a language used within the LLVM project for generating all kinds of files,
+when maintaining those files by hand would be very difficult.
+
+For example, it is used to define all of the instructions that can be used on a
+particular architecture. The information is defined once in TableGen and we can
+produce many things based on it. C++ code, documentation, command line options,
+etc.
```

Comment:
based on it. For example, C++..... and so on

(avoid Latin contractions - not good for non-native speakers)

Reply:
I've replaced all the etc. and e.g. uses.